### PR TITLE
Add tail and follow support to logs [full ci]

### LINKED ISSUE #1682: Datastore file streaming
In support of #366 - we need to figure out a way to stream log files from the datastore.  We are currently able to download a complete file, but we'll want to "tail" the file to support docker logs streaming.

### PR DESCRIPTION
**Pull request by dougm**
Fixes #1682

### CODE REVIEW

File: lib/portlayer/exec/container.go [hunk 1]
```diff
@@ -377,15 +382,23 @@ func (c *Container) Signal(ctx context.Context, num int64) error {
 	return c.startGuestProgram(ctx, "kill", fmt.Sprintf("%d", num))
 }
 
-func (c *Container) LogReader(ctx context.Context) (io.ReadCloser, error) {
+func (c *Container) stopped() {
```
Comment (sflxn) [inline lib/portlayer/exec/container.go]: Looks like this only deals with the log reader.  Why not call it stopLogReaders() or something?  Not a big deal.

**Comment by sflxn** [comment]
Just a naming comment.  Other then that, LGTM.

File: lib/portlayer/exec/container.go (hunk 1 above)
Comment (cgtexmex) [inline lib/portlayer/exec/container.go]: agree that a different function name would be preferable

Comment (dougm) [inline lib/portlayer/exec/container.go]: Just wanted a generic method that'd be called when an container is stopped, to
avoid sandwiching duplicate code into the same two places in the future.  I can rename it.

Comment (cgtexmex) [inline lib/portlayer/exec/container.go]: @dougm I like the idea of a generic stop method that could be called for any clean up routines....I'd name it onStop and have the state evaluation in the function

Comment (dougm) [inline lib/portlayer/exec/container.go]: ok onStop it is.  Why would 'onStop' evaluate the state?  eventCallback could probably be refactored a bit, but that's where we are already evaluating state changes and dispatching based on the state change.

Comment (cgtexmex) [inline lib/portlayer/exec/container.go]: eventCallback will only execute onStop if the event is stopping the container -- so out of band stops / power offs, etc...or a short lived container that exits and powers down.  Executing a client side stop will not.  

So thinking potentially others might call onStop and didn't want to recreate the if statement in those places...no biggie...so whatever you feel best about

Comment (dougm) [inline lib/portlayer/exec/container.go]: Right, I added to call there to handle out-of-band power off and container shutting down on its own.  The Container.stop() method has an explicit call to Container.onStop to handle client stop call, in which case there's no need to check the state.

**Comment by cgtexmex** [comment]
LGTM

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Comment by dougm** [comment]
♻️

**Pull request merged**